Dart/src/com/jetbrains/lang/dart/ide/findUsages/DartComponentUsageGroup.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.findUsages;
+
+import com.google.dart.server.utilities.general.ObjectUtilities;
+import com.intellij.navigation.NavigationItemFileStatus;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.DataProvider;
+import com.intellij.openapi.util.Iconable;
+import com.intellij.openapi.vcs.FileStatus;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.SmartPointerManager;
+import com.intellij.psi.SmartPsiElementPointer;
+import com.intellij.usageView.UsageInfo;
+import com.intellij.usages.UsageGroup;
+import com.intellij.usages.UsageView;
+import com.jetbrains.lang.dart.psi.DartComponent;
+import com.jetbrains.lang.dart.psi.DartComponentName;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+
+class DartComponentUsageGroup implements UsageGroup, DataProvider {
```

Comment:
PsiNamedElementUsageGroupBase looks pretty much the same, but there are two issues with it.

1. It implements calcData(), but does not implement TypeSafeDataProvider, so calcData() is not used. This means that Preview does not work.

2. If I make it implement TypeSafeDataProvider, it still does not work well. It does not highlight names, but instead just a first token.
![image](https://cloud.githubusercontent.com/assets/384794/9012284/7b04e708-3768-11e5-935e-5103f0f19c8f.png)
![image](https://cloud.githubusercontent.com/assets/384794/9012298/878cc55e-3768-11e5-94da-23b12e188670.png)

So, at least for now, DartComponentUsageGroup seems to provide better functionality.